openedx/core/djangoapps/global_statistics/tasks.py

```diff
@@ -0,0 +1,52 @@
+"""
+This file contains periodic tasks for global_statistics, which will collect data about Open eDX users
+and send this data to appropriate service for further processing.
+"""
+import json
+import requests
+
+from celery.task import task
+
+from lms.envs.devstack_with_worker import PERIODIC_TASK_POST_URL, PLATFORM_LATITUDE, PLATFORM_LONGITUDE, SITE_NAME
+from xmodule.modulestore.django import modulestore
+from student.models import UserProfile
+from .models import TokenStorage
+
+
+@task
+def count_data():
+    """
+    Periodic task function that gathers information about the students amount,
+    geographical coordinates of the platform, courses amount and
+    makes a POST request with the data to the appropriate service.
+    """
+
+    # Get IP address of the platform and convert it to latitude, longitude.
+    check_ip_url = 'http://freegeoip.net/json'
+    ip_data = requests.get(check_ip_url)
+    ip_data_json = json.loads(ip_data.text)
+    if PLATFORM_LATITUDE and PLATFORM_LONGITUDE:
+        latitude = PLATFORM_LATITUDE
+        longitude = PLATFORM_LONGITUDE
+    else:
+        latitude = ip_data_json['latitude']
+        longitude = ip_data_json['longitude']
+
+    # Get students amount within current platform.
+    students_amount = UserProfile.objects.count()
+
+    # Get courses amount within current platform.
+    courses_amount = len(modulestore().get_courses())
+
+    # Secret token to authorize our platform on remote server.
+    secret_token, created = TokenStorage.objects.get_or_create(pk=1)
```

Comment:
Actually authorization is handled on "acceptor" server side, I mean the first time edx-platform sends data to our remote "acceptor" server, we generate and save(register) our key on "acceptor" side. Only after that "acceptor" sends this key back to edx-platform for further data exchange abilities. 
We need `create`, because the first time edx-platform sends its data to remote "acceptor" server, edx has nor token no any object in it's DB. So instead of write something like: 
```
try:
    retrieve object from DB
except:
    object does not exists
```
I decided to create initial object and save it as a null value to edx's DB. In any case this object will be overwritten to the appropriate token when "acceptor" sends back generated key to the platform.